apoc.date.parse gives error when Date is NULL or empty string

Using `apoc.date.parse` and when date value NULL or empty string is passed it gives the following error

```
with "" as dateStr 
with apoc.date.parse(dateStr,"s","yyyy-MM-dd") as dt return dt
```

Error:

```
Neo.ClientError.Procedure.ProcedureCallFailed
Neo.ClientError.Procedure.ProcedureCallFailed: Failed to invoke function `apoc.date.parse`: Caused by: java.text.ParseException: Unparseable date: ""
```

## Expected Behavior (Mandatory)

Expected behavior - I was expecting NULL or empty string from the function if the date passed is NULL or empty String. 

Currently used versions: Neo4j version 3.5.17
APOC version 3.5.0.11


##Workaround as suggested by Michael 

with "" as dateStr 
with case dateStr  WHEN ""  then "" else apoc.date.parse(dateStr,"s","yyyy-MM-dd") end as dt return dt

Returns: 

```
╒════╕
│"dt"│
╞════╡
│""  │
└────┘
```

--- a/core/src/main/java/apoc/date/Date.java
+++ b/core/src/main/java/apoc/date/Date.java
@@ -2,6 +2,7 @@ package apoc.date;
 
 import apoc.util.DateFormatUtil;
 import apoc.util.Util;
+import org.apache.commons.lang3.StringUtils;
 import org.neo4j.graphdb.Label;
 import org.neo4j.graphdb.Node;
 import org.neo4j.procedure.*;
@@ -162,7 +163,7 @@ public class Date {
 	@UserFunction
 	@Description("apoc.date.parse('2012-12-23','ms|s|m|h|d','yyyy-MM-dd') - parse date string using the specified format into the specified time unit")
 	public Long parse(@Name("time") String time, @Name(value = "unit", defaultValue = "ms") String unit, @Name(value = "format",defaultValue = DEFAULT_FORMAT) String format, final @Name(value = "timezone", defaultValue = "") String timezone) {
-		Long value = parseOrThrow(time, getFormat(format, timezone));
+		Long value = StringUtils.isBlank(time) ? null : parseOrThrow(time, getFormat(format, timezone));
 		return value == null ? null : unit(unit).convert(value, TimeUnit.MILLISECONDS);
 	}
 

--- a/core/src/test/java/apoc/date/DateTest.java
+++ b/core/src/test/java/apoc/date/DateTest.java
@@ -112,6 +112,11 @@ public class DateTest {
 				row -> assertNull(row.get("value")));
 	}
 
+	@Test 
+	public void testToUnixtimeWithEmptyInput() throws Exception {
+		testCall(db, "RETURN apoc.date.parse(' ','s') AS value", row -> assertNull(row.get("value")));
+	}
+
 	@Test public void testFromUnixtime() throws Exception {
 		testCall(db,
 				"RETURN apoc.date.format(0,'s') AS value",

